### PR TITLE
CDS-57109-auto-reject-previous-approvals

### PR DESCRIPTION
**Pull request by roshnisarangadharan**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by shylajasun — APPROVED** [review]

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md [hunk 1]
```diff
@@ -84,11 +84,21 @@ The maximum timeout duration is 24 days.The timeout countdown appears when the s
 1. In **Approval Message**, add the message for the users in **Approvers**.
 
 2. Enable the **Include Pipeline execution history in approval details** option to show approvers the pipeline's execution history. This can help an approver compare the current execution info with historical data.
+3. Enable the **Auto-Reject previous deployments paused in this step on approval** option to reject old executions waiting for approval when a latest step is approved. 
+   
+   :::info Limitations:
+
+   * If you have two approval steps in a step group of a stage with the same step identifier. Harness won't be able to differentiate between the approval steps in such case, and rejects previous deployments with the same identifier.
```
Comment (roshnisarangadharan) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:91]:
```suggestion
   * If you have two approval steps in a step group of a stage with the same step identifier. Harness won't be able to differentiate between the approval steps, and rejects previous deployments with the same identifier.
```

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 2]
```diff
@@ -83,26 +87,38 @@ The default timeout for an Approval step is **1d** (24 hours). You can use `**w
...
 Enable this option to provide approvers with the execution history for this Pipeline. This can help approvers make their decision.
 
-### Step 3: Select Approvers
+### Auto-reject previous deployments in approval
+
+Enable this option to reject old executions waiting for approval when a latest step is approved. 
+
+:::info
+
+If you have two approval steps in a step group of a stage with the same step identifier. Harness won't be able to differentiate between the approval steps in such case, and rejects previous deployments with the same identifier.
```
Comment (roshnisarangadharan) [inline docs/platform/9_Approvals/adding-harness-approval-stages.md:105]:
```suggestion
If you have two approval steps in a step group of a stage with the same step identifier. Harness won't be able to differentiate between the approval steps, and rejects previous deployments with the same identifier.
```

**Review by Y0709 — APPROVED** [review]

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md [hunk 3]
```diff
@@ -84,11 +84,21 @@ The maximum timeout duration is 24 days.The timeout countdown appears when the s
 1. In **Approval Message**, add the message for the users in **Approvers**.
 
 2. Enable the **Include Pipeline execution history in approval details** option to show approvers the pipeline's execution history. This can help an approver compare the current execution info with historical data.
+3. Enable the **Auto-Reject previous deployments paused in this step on approval** option to reject old executions waiting for approval when a latest step is approved. 
+   
+   :::info Limitations:
+
+   * If you have two approval steps in a step group of a stage with the same step identifier. Harness won't be able to differentiate between the approval steps, and rejects previous deployments with the same identifier.
```
Comment (StrawberryPad) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/approvals/using-harness-approval-steps-in-cd-stages.md:91]:
```suggestion
   * If you have two approval steps in a step group of a stage with the same step identifier, Harness won't be able to differentiate between the approval steps, and rejects previous deployments with the same identifier.
```

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 4]
```diff
@@ -64,7 +68,7 @@ YAML Example
                                   defaultValue: myvalue  
       failureStrategies: []
 ```
-### Step 1: Add Approval Stage
+### Add the Approval stage
```
Comment (StrawberryPad) [inline docs/platform/9_Approvals/adding-harness-approval-stages.md:71]:
```suggestion
### Add the approval stage
```

File: docs/platform/9_Approvals/adding-harness-approval-stages.md [hunk 5]
```diff
@@ -83,26 +87,38 @@ The default timeout for an Approval step is **1d** (24 hours). You can use `**w
...
 Enable this option to provide approvers with the execution history for this Pipeline. This can help approvers make their decision.
 
-### Step 3: Select Approvers
+### Auto-reject previous deployments in approval
+
+Enable this option to reject old executions waiting for approval when a latest step is approved. 
+
+:::info
+
+If you have two approval steps in a step group of a stage with the same step identifier. Harness won't be able to differentiate between the approval steps, and rejects previous deployments with the same identifier.
```
Comment (StrawberryPad) [inline docs/platform/9_Approvals/adding-harness-approval-stages.md:105]:
```suggestion
If you have two approval steps in a step group of a stage with the same step identifier, Harness won't be able to differentiate between the approval steps, and rejects previous deployments with the same identifier.
```

**Review by StrawberryPad — CHANGES_REQUESTED** [review]

File: docs/platform/9_Approvals/adding-harness-approval-stages.md (hunk 4 above)
Comment (roshnisarangadharan) [reply docs/platform/9_Approvals/adding-harness-approval-stages.md:71]: using title case for names of steps and stages

**Pull request merged**